### PR TITLE
Fix movable construction

### PR DESCRIPTION
**Pull request by kanstantsin-chernik**
Details:
- Fix of pool initialization with passed args #240
- Test to check move of shared scope

Reviewers:
@krzysztof-jusiak

### CODE REVIEW

File: include/boost/di.hpp [hunk 1]
```diff
@@ -2712,16 +2712,16 @@ class injector<TConfig, pool<>, TDeps...> : injector_base, pool<bindings_t<TDeps
  private:
   template <class... TArgs>
   explicit injector(const from_deps&, TArgs... args) noexcept
-      : pool_t{copyable_t<deps>{}, core::pool_t<TArgs...>{static_cast<TArgs&&>(args)...}} {}
```
Comment (kris-jusiak) [inline include/boost/di.hpp:2715]: I think that the copyable can be removed then?

File: include/boost/di/core/injector.hpp [hunk 2]
```diff
@@ -269,17 +269,17 @@ class injector __BOOST_DI_CORE_INJECTOR_POLICY()(<TConfig, pool<>, TDeps...>) :
  private:
   template <class... TArgs>
   explicit injector(const from_deps&, TArgs... args) noexcept
-      : pool_t{copyable_t<deps>{}, core::pool_t<TArgs...>{static_cast<TArgs&&>(args)...}} {}
+      : pool_t{bindings_t<TArgs...>{}, core::pool_t<TArgs...>{static_cast<TArgs&&>(args)...}} {}
```
Comment (kris-jusiak) [inline include/boost/di/core/injector.hpp:257]: Not for now, but I think we can simplify it by removing `bindings_t` and add templated move constructor for the pool from another pool?

```cpp
  template <class... Ts>
  pool(pool<Ts...>&& p) noexcept : pool(static_cast<Ts&&>(p)...) { }
```

Comment (kanstantsin-chernik) [reply include/boost/di/core/injector.hpp:257]: I think so, but it has quite tricky logic behind. I did my best not to break anything

File: include/boost/di.hpp (hunk 1 above)
Comment (kanstantsin-chernik) [reply include/boost/di.hpp:2715]: Good point. Thank you for noticing!

**Pull request merged**